runtime/sema/type.go

```diff
@@ -6309,6 +6307,78 @@ func (t *ReferenceType) Resolve(typeArguments *TypeParameterTypeOrderedMap) Type
 	}
 }
 
+func (t *ReferenceType) GetMembers() map[string]MemberResolver {
+	t.initializeMemberResolvers()
+	return t.memberResolvers
+}
+
+const ReferenceTypeDereferenceFunctionName = "dereference"
+
+const referenceTypeDereferenceFunctionDocString = `
+	Returns a copy of the reference value after dereferencing.
```

Comment:
```suggestion
	Returns a copy of the referenced value after dereferencing.
```

Reply:
IMO we should also add something to doc string about how this only works for primitive values